arangosh/Benchmark/BenchmarkThread.h

```diff
@@ -75,9 +77,85 @@ class BenchmarkThread : public arangodb::Thread {
         _counter(0),
         _time(0.0),
         _errorHeader(basics::StringUtils::tolower(StaticStrings::Errors)),
-        _verbose(verbose) {}
+        _verbose(verbose),
+        _minTime(-1.0),
+        _maxTime(0.0),
+        _avgTime(0.0),
+        _histogrammScope(0.0),
+        _histogramm(1000, 0) { }
 
   ~BenchmarkThread() { shutdown(); }
+  //////////////////////////////////////////////////////////////////////////////
+  /// @brief add one time
+  //////////////////////////////////////////////////////////////////////////////
+
+  void oneTime(double time) {
+    if (_minTime == -1.0 || time < _minTime) {
+      _minTime = time;
+    }
+  
+    if (time > _maxTime) {
+      _maxTime = time;
+    }
+
+    _avgTime = ((_avgTime * _counter) + time) / (_counter + 1);
```

Comment:
I think this is not ideal, as it will continuously improve the precision of `_avgTime`.
Why not keep a seperate sum of times plus a counter, and in the very end do a division or sum/count?

Reply:
hm, i.e. collecd do it that way, and its not overflowing like this - plus you always have the average at hand?

Reply:
I agree with @jsteemann. Why not use an exponential moving average. That would allow to check if things get better over time. Now we have the situation, that we can not see improvements. Obviously it really depends if the counting is a continuous thing or not. Only if it is a one-shot thing the current solution is what we want. Otherwise it should be changed.